app/models/user.rb

```diff
@@ -37,11 +37,18 @@ def get_coords
   end
 
   def set_profile_picture
-    image = ImageService.new.user_image
-    if !image[:error]
-      self.profile_picture = image[:data][:attributes][:raw_image]
+    begin
+      image = ImageService.new.user_image
+    rescue Faraday::ConnectionFailed
+      RetryImageServiceJob.perform_in(15.seconds, self.id)
```

Comment:
AH I see, if there is some type of connection failure, we can just setup that background worker to work immediately and even set the time for it. How cool!